lsp-mode.el

```diff
@@ -2864,7 +2864,7 @@ Applies on type formatting."
 
 (defun lsp-workspace-root (&optional path)
   "Find the workspace root for the current file or PATH."
-  (when-let (file-name (or path (buffer-file-name)))
+  (when-let (file-name (f-canonical (or path (buffer-file-name))))
```

Comment:
f-canonical will throw if called with nil

Reply:
I'll update to handle the nil case

Reply:
Just checked and f-ancestor-of? throws when file-name is nil so this change doesn't change ny behavior here